File: HomeAutomation/src/RemoteAccessApp.java
public class RemoteAccessApp implements IObserver {
    private HomeState homeState;
    private String room;


    public RemoteAccessApp(HomeState homeState, String room) {
        this.homeState = homeState;
        this.room = room;
        homeState.registerObserver(this);
    }

    @Override
    public void update(String stateName, Object value) {

        }
    }

File: HomeAutomation/src/src/RemoteAccessApp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace src
{
    public class RemoteAccessApp : IObserver
    {
        private CoreComponent coreComponent;

        public RemoteAccessApp(CoreComponent coreComponent)
        {
            this.coreComponent = coreComponent;
            coreComponent.RegisterObserver(this);
        }

        public void Update(string key, string value)
        {
            if (key == "notification")
            {
                GenerateNotification(value);
            }
        }

        public void GenerateNotification(string message)
        {
            Console.WriteLine($"Notification: {message}");
        }

        public void DisplayState()
        {
            Console.WriteLine("Displaying current state...");
        }

        public void RemoteControl(string command)
        {
            coreComponent.SetState("notification", command);
        }
    }
}
